Test Failure: PolicyExecutorTest.testStartTimeout expected 0 but was 2160672 ns

Test Failure: com.ibm.ws.threading.policy.PolicyExecutorTest.testStartTimeout

```
testStartTimeout:junit.framework.AssertionFailedError: 2019-05-01-19:37:52:065 The response did not contain [SUCCESS].  Full output is:
ERROR: Caught exception attempting to call test method testStartTimeout on servlet web.PolicyExecutorServlet
java.lang.AssertionError: expected:<0> but was:<2160672>
at web.PolicyExecutorServlet.testStartTimeout(PolicyExecutorServlet.java:4553)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.app.FATServlet.doGet(FATServlet.java:71)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1221)
at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1005)
at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:923)
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1061)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:417)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:376)
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpIn...
```

The startTimeout is reached and two threads simultaneously attempt to abort the task, overwriting the 

One of those is a thread from the global thread pool which has just finished the previous task and is looking for the next, which it finds to have exceeded its start timeout:
```
com.ibm.ws.threading.StartTimeoutException: CWWKE1205E: The PolicyExecutorProvider-testStartTimeout executor was unable to start the web.SharedIncrementTask@ede17d9d task after 305,387,379 nanoseconds because the task exceeded its startTimeout of 300,000,000 nanoseconds.
	at com.ibm.ws.threading.internal.PolicyExecutorImpl$GlobalPoolTask.run(PolicyExecutorImpl.java:193)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:825)
```
before which the following happens:
```
            long nsQueueEnd = System.nanoTime();
            ...
                } else { // timed out
                    next.nsRunEnd = next.nsQueueEnd = nsQueueEnd;
                    next.abort(false, new StartTimeoutException(next.getIdentifier(), next.getTaskName(), //
                                    nsQueueEnd - next.nsAcceptBegin, //
                                    next.nsStartBy - next.nsAcceptBegin));
                }
```

The other is the test case doing future.get > await > abort
```
            nsRunEnd = nsQueueEnd = nsAcceptEnd = System.nanoTime();
or
                    nsRunEnd = nsQueueEnd = System.nanoTime();
```

where the interleaving of threads makes it possible for nsRunEnd and nsQueueEnd to get out of sync.

--- a/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/PolicyTaskFutureImpl.java
+++ b/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/PolicyTaskFutureImpl.java
@@ -643,7 +643,11 @@ public class PolicyTaskFutureImpl<T> implements PolicyTaskFuture<T> {
     public final long getElapsedRunTime(TimeUnit unit) {
         long begin = nsQueueEnd;
         long elapsed = nsRunEnd - begin;
-        return unit.convert(elapsed >= 0 ? elapsed : begin - nsAcceptBegin > 0 ? System.nanoTime() - begin : 0, TimeUnit.NANOSECONDS);
+        if (elapsed < 0)
+            elapsed = begin - nsAcceptBegin > 0 ? System.nanoTime() - begin : 0;
+        else if (elapsed > 0 && state.get() == ABORTED)
+            elapsed = 0; // nsQueueEnd,nsRunEnd can get out of sync when abort is attempted by multiple threads at once
+        return unit.convert(elapsed, TimeUnit.NANOSECONDS);
     }
 
     @Trivial

